gyunuk/composestudy/app/src/main/java/com/bendeng/composestudy/ui/screen/home/HomeScreen.kt

```diff
@@ -0,0 +1,205 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.bendeng.composestudy.R
+import com.bendeng.composestudy.domain.model.DocumentData
+import com.bendeng.composestudy.ui.components.LoadingDialog
+import com.bendeng.composestudy.ui.components.RoundedGlideImage
+import com.bendeng.composestudy.ui.util.DateTimeUtils
+
+@Composable
+fun HomeScreen(
+    modifier: Modifier = Modifier,
+    viewModel: HomeViewModel = hiltViewModel(),
+) {
+    val state by viewModel.container.stateFlow.collectAsState()
+    val context = LocalContext.current
+
+    LaunchedEffect(viewModel) {
+        viewModel.container.sideEffectFlow.collect { sideEffect ->
+            when (sideEffect) {
+                is HomeSideEffect.Toast -> Toast.makeText(
+                    context,
+                    sideEffect.text,
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+        }
+    }
+
+    Surface(
+        modifier = modifier.padding(start = 20.dp, end = 20.dp, top = 20.dp)
+    ) {
+        Column(modifier = modifier) {
```

Comment:
```suggestion
        Column(modifier = Modifier) {
```
modifier 수정자는 하나의 레이아웃(컴포저블 함수)에서만 진행하는 것이 좋습니다! 예기치 않은 맞춤설정에 Surface 에 들어가야하는 설정이 Column에 들어갈 수 있는 상황이 발생할 수 있어요!!!

Reply:
와 안그래도 이 부분에서 학습하면서 엄청 헷갈렸는데 compose 조언 너무 감사합니다😀